R/run-region-updates.R

```diff
@@ -179,19 +153,153 @@ rru_log_outcome <- function(outcome) {
+      futile.logger::flog.trace("dataset %s has a complete results set", dataset_name)
+      dataset_status <- "Partial"
+      dataset_processed <- TRUE
+    }else if (dataset_counts$failures == 0) {
+      futile.logger::flog.trace("dataset %s has a completely timed out", dataset_name)
+      dataset_status <- "Timed Out"
+    }else {
+      futile.logger::flog.trace("dataset %s had an error", dataset_name)
+      dataset_status <- "Error"
+    }
+    if (nrow(status_row) == 0) {
+      futile.logger::flog.trace("no status record exists for %s so create a new one", dataset_name)
+      status <- dplyr::rows_insert(
+        status,
+        data.frame(
+          dataset = dataset_name,
+          last_attempt = outcome[[dataset_name]]$start,
+          last_attempt_status = dataset_status,
+          latest_results = dplyr::if_else(dataset_processed, outcome[[dataset_name]]$start, NULL),
+          latest_results_status = ifelse(dataset_processed, dataset_status, NULL),
+          latest_results_data_up_to = dplyr::if_else(dataset_processed, outcome[[dataset_name]]$max_data_date, NULL),
+          latest_results_successful_regions = ifelse(dataset_processed, dataset_counts$successes, 0),
+          latest_results_timing_out_regions = ifelse(dataset_processed, dataset_counts$timeouts, 0),
+          latest_results_failing_regions = ifelse(dataset_processed, dataset_counts$failures, 0),
+          oldest_region_results = outcome[[dataset_name]]$oldest_results
+        ),
+        by = c("dataset")
+      )
+    } else {
+      futile.logger::flog.trace("status record exists for %s", dataset_name)
+      status_row$dataset <- dataset_name
+      status_row$last_attempt <- outcome[[dataset_name]]$start
+      status_row$last_attempt_status <- dataset_status
+      status_row$latest_results <- dplyr::if_else(dataset_processed, outcome[[dataset_name]]$start, status_row$latest_results)
+      status_row$latest_results_status <- ifelse(dataset_processed, dataset_status, status_row$latest_results_status)
+      status_row$latest_results_data_up_to <- dplyr::if_else(dataset_processed, outcome[[dataset_name]]$max_data_date, status_row$latest_results_data_up_to)
+      status_row$latest_results_successful_regions <- ifelse(dataset_processed, dataset_counts$successes, status_row$latest_results_successful_regions)
+      status_row$latest_results_timing_out_regions <- ifelse(dataset_processed, dataset_counts$timeouts, status_row$latest_results_timing_out_regions)
+      status_row$latest_results_failing_regions <- ifelse(dataset_processed, dataset_counts$failures, status_row$latest_results_failing_regions)
+      status_row$oldest_region_results <- outcome[[dataset_name]]$oldest_results
+      status <- dplyr::rows_upsert(status, status_row, by = c("dataset"))
+    }
   }
   futile.logger::flog.trace("writing file")
-  write.csv(stats, file = filename, row.names = FALSE)
+  write.csv(stats, file = stats_filename, row.names = FALSE)
```

Comment:
@JAllen42 how hard would it be to sort these before saving them? Region / subregion alphabetically would be awesome ;)